test/jdk/javax/swing/text/html/StyleSheet/8260687/BodyInheritedFontSize.java

```diff
@@ -0,0 +1,168 @@
+            frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+            frame.add(new JScrollPane(htmlPane), BorderLayout.CENTER);
+            frame.setLocationRelativeTo(null);
+            frame.pack();
+            frame.setVisible(true);
+        }
+
+        // Ignore the result but perform layout
+        htmlPane.getPreferredSize();
+
+        return htmlPane;
+    }
+
+    public static void main(String[] args) throws Exception {
+        final List<String> argsList = Arrays.asList(args);
+        final boolean showFrame = toShowFrame(argsList);
+        final boolean debugPrint = toDebugPrint(argsList);
+
+        final List<Exception> exceptions = new ArrayList<>(2);
+        SwingUtilities.invokeAndWait(() -> {
+            for (boolean w3cUnits : new boolean[] {true, false}) {
+                JEditorPane htmlPane = createEditorPane(w3cUnits, showFrame);
+                try {
+                    checkFontSize(htmlPane, w3cUnits, debugPrint);
+                } catch (Exception e) {
+                    exceptions.add(e);
+                }
+            }
+        });
+        if (exceptions.size() > 0) {
+            exceptions.forEach(System.err::println);
+            throw new RuntimeException(
+                    "Test failed: " + exceptions.get(0).getMessage(),
+                    exceptions.get(0));
+        }
+    }
+
+    private static boolean toShowFrame(final List<String> argsList) {
+        return argsList.contains("-show");
+    }
+
+    private static boolean toDebugPrint(final List<String> argsList) {
+        return argsList.contains("-print");
+    }
+
+    private static void checkFontSize(JEditorPane htmlPane,
+                                      boolean w3cUnits,
+                                      boolean debugPrint) {
+        final View rootView = htmlPane.getUI().getRootView(htmlPane);
+        final View boxView = rootView.getView(0);
+        final View bodyView = boxView.getView(1);
+
+        int fontSizePrecentage = getViewFontSize(bodyView.getView(0), debugPrint);
+        int fontSizeInherited  = getViewFontSize(bodyView.getView(1), debugPrint);
+        int fontSizeExplicit   = getViewFontSize(bodyView.getView(2), debugPrint);
+        if (debugPrint) {
+            System.out.println("w3cUnits: " + w3cUnits + "\n"
+                    + "Percentage: " + fontSizePrecentage + "\n"
+                    + "Inherited: " + fontSizeInherited + "\n"
+                    + "Explicit: " + fontSizeExplicit + "\n");
+        }
+        if (fontSizeInherited != fontSizeExplicit) {
+            throw new RuntimeException("The font size is different with "
+                    + (w3cUnits ? "w3cUnits" : "stdUnits") + ": "
+                    + "Percentage: " + fontSizePrecentage + " vs. "
+                    + "Inherited: " + fontSizeInherited + " vs. "
+                    + "Explicit: " + fontSizeExplicit);
+        }
```

Comment:
I compared two pairs: `fontSizeInherited != fontSizeExplicit` and `fontSizePrecentage != fontSizeExplicit`. These two checks also imply `fontSizeInherited == fontSizePrecentage` which is also what we want.

Comparing only one pair of the three items is not enough.

Typo: fontSizePrecentage → fontSizePercentage.